### PR TITLE
Mqtt add object

### PR DESCRIPTION
**Pull request by Smeagolworms4**
## Proposed change

Add documentation of object_id in relation with with PR: https://github.com/home-assistant/core/pull/58728

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:  https://github.com/home-assistant/core/pull/58728
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by emontnemery** [comment]
Each MQTT entity's documentation should be updated with the new configuration parameter, not just the discovery starting page.

**Review by emontnemery — CHANGES_REQUESTED** [review]
Each MQTT entity's documentation should be updated with the new configuration parameter, not just the discovery starting page.

File: source/_docs/mqtt/discovery.markdown [hunk 1]
```diff
@@ -481,3 +482,19 @@ If the device supports gps coordinates then they can be sent to Home Assistant b
   "gps_accuracy": 1.2
  }
 ```
+
+### Used object_id to generate the entity id
+
+The entity identifier is generated from the device name. If you defined an object_id then it will be used instead.
+- Configuration topic: `homeassistant/sensor/devive1/config`
```
Comment (jbouwh) [inline source/_docs/mqtt/discovery.markdown]: Spelling device

File: source/_docs/mqtt/discovery.markdown [hunk 2]
```diff
@@ -481,3 +482,19 @@ If the device supports gps coordinates then they can be sent to Home Assistant b
...
+- Example configuration payload:
+
+```json
+{
+  "name":"My Super Device",
+  "object_id":"device1",
+  "state_topic": "homeassistant/sensor/devive1/state",
+ }
```
Comment (jbouwh) [inline source/_docs/mqtt/discovery.markdown:499]: Same here

**Review by jbouwh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**